### PR TITLE
Add staking commands to CLI

### LINKED ISSUE #1943: Add staking operations to CLI

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds the CLI staking commands.

## Links to any relevant issues

Closes #1943

### CODE REVIEW

File: cli/src/wallet_cli/mod.rs [hunk 1]
```diff
@@ -428,6 +465,37 @@ pub async fn balance_command(wallet: &Wallet) -> Result<(), Error> {
...
 
+// `begin-staking` command
+pub async fn begin_staking_command(
+    wallet: &Wallet,
+    account_id: AccountId,
+    staked_amount: u64,
+    fixed_cost: u64,
+    staking_period: Option<u32>,
+) -> Result<(), Error> {
+    println_log_info!("Beginning a staking for {account_id}.");
```
Comment (Thoralf-M) [inline cli/src/wallet_cli/mod.rs]:
```suggestion
    println_log_info!("Begin staking for {account_id}.");
```

File: cli/src/wallet_cli/mod.rs [hunk 2]
```diff
@@ -1292,7 +1394,16 @@ pub async fn prompt_internal(
                             ensure_password(wallet).await?;
                             allot_mana_command(wallet, mana, account_id).await
                         }
+                        WalletCommand::AnnounceCandidacy { account_id } => {
```
Comment (thibault-martinez) [inline cli/src/wallet_cli/mod.rs:1397]: I believe you have to call `ensure_password(wallet).await?;` before all operations that sends something

Comment (DaughterOfMars) [reply cli/src/wallet_cli/mod.rs:1397]: good to know

Comment (DaughterOfMars) [reply cli/src/wallet_cli/mod.rs:1397]: done

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**